app/Models/Entry.php

```diff
@@ -636,21 +636,41 @@ public function matches(FreshRSS_BooleanSearch $booleanSearch): bool {
 						$ok &= stripos(implode(';', $this->authors), $author) !== false;
 					}
 				}
+				if ($ok && $filter->getAuthorRegex()) {
+					foreach ($filter->getAuthorRegex() as $author) {
+						$ok &= preg_match($author, implode("\n", $this->authors)) === 1;
+					}
+				}
 				if ($ok && $filter->getNotAuthor()) {
 					foreach ($filter->getNotAuthor() as $author) {
 						$ok &= stripos(implode(';', $this->authors), $author) === false;
 					}
 				}
+				if ($ok && $filter->getNotAuthorRegex()) {
+					foreach ($filter->getNotAuthorRegex() as $author) {
+						$ok &= preg_match($author, implode("\n", $this->authors)) === 0;
+					}
+				}
 				if ($ok && $filter->getIntitle()) {
 					foreach ($filter->getIntitle() as $title) {
 						$ok &= stripos($this->title, $title) !== false;
 					}
 				}
+				if ($ok && $filter->getIntitleRegex()) {
+					foreach ($filter->getIntitleRegex() as $title) {
+						$ok &= preg_match($title, $this->title) === 1;
+					}
+				}
 				if ($ok && $filter->getNotIntitle()) {
 					foreach ($filter->getNotIntitle() as $title) {
 						$ok &= stripos($this->title, $title) === false;
 					}
 				}
+				if ($ok && $filter->getNotIntitleRegex()) {
+					foreach ($filter->getNotIntitleRegex() as $title) {
+						$ok &= preg_match($title, $this->title) === 0;
+					}
+				}
 				if ($ok && $filter->getTags()) {
```

Comment:
It looks like it would be relatively easy to add tag regex support too. I know I've had cases where I wanted to match a list of tags and it's a a lot easier to do `#/\(tag1|tag2|tag3|tag4|tag5\)/i` than it is to do `#tag1 OR #tag2 OR #tag3 OR #tag4 OR #tag5`, especially if there are some shared sub-string between each of those tags.

Reply:
Done. Tests welcome

Reply:
P.S. In multiline